packages/cysync-core/src/bgTask/walletSyncTask/index.tsx

```diff
@@ -1,21 +1,30 @@
 import React, { useEffect } from 'react';
 
-import { syncWalletsWithDevice } from '~/actions';
+import { openWalletSyncErrorDialog, syncWalletsWithDevice } from '~/actions';
 import { useDevice } from '~/context';
-import { useAppDispatch } from '~/store';
+import { selectWallets, useAppDispatch, useAppSelector } from '~/store';
 
 export const WalletSyncTask: React.FC = () => {
   const dispatch = useAppDispatch();
+  const { deletedWallets } = useAppSelector(selectWallets);
 
   const { connection } = useDevice();
 
   const onConnectionChange = async () => {
     dispatch(syncWalletsWithDevice(connection));
   };
 
+  const onDeleteChange = () => {
+    dispatch(openWalletSyncErrorDialog);
+  };
+
   useEffect(() => {
     onConnectionChange();
   }, [connection]);
 
+  useEffect(() => {
+    if (deletedWallets.length > 0) onDeleteChange();
+  }, [deletedWallets]);
+
```

Comment:
Should we close the dialog if the `deletedWallets`  are 0?

Reply:
We haven't opened the dialog yet.